src/renderer/components/reader/ReaderMenu.tsx

```diff
@@ -195,9 +219,55 @@ export class ReaderMenu extends React.Component<Props, State> {
         return undefined;
     }
 
+    private buildGoToPageSection() {
+        const { __ } = this.props;
+        const error = this.state.pageError;
+        return <div className={styles.goToPage}>
+            <p className={styles.title}>{__("reader.navigation.goToTitle")}</p>
+            {error &&
+                <p
+                    className={styles.goToErrorMessage}
+                    aria-live="assertive"
+                    aria-relevant="all"
+                    role="alert"
+                >
+                    { __("reader.navigation.goToError") }
+                </p>
+            }
+            <form onSubmit={this.handleSubmitPage}>
+                <input
+                    ref={(ref) => this.goToRef = ref}
+                    type="number"
+                    aria-invalid={error}
+                    disabled={!this.props.publication.PageList}
+                    placeholder={__("reader.navigation.goToPlaceHolder")}
+                    alt={__("reader.navigation.goToPlaceHolder")}
+                />
+                <button
+                    type="submit"
+                    disabled={!this.props.publication.PageList}
+                >
+                    { __("reader.navigation.goTo") }
+                </button>
+            </form>
+        </div>;
+    }
+
     private closeBookarkEditForm() {
         this.setState({ bookmarkToUpdate: undefined });
     }
+
+    private handleSubmitPage(e: any) {
+        e.preventDefault();
+        const pageNbr = this.goToRef.value;
+        const foundPage = this.props.publication.PageList.find((page) => page.Title === pageNbr.toString());
```

Comment:
`page` is in fact a `Link` object. `page.Title` is a "cleaned-up" string extracted from the text descendants of the corresponding `<a>` HTML element in the EPUB3 Navigation Document (`<nav epub:type="pagelist"`). The "cleanup" method is: `pageLink.Title = aElement.textContent.trim().replace(/\s\s+/g, " ")`.

`pageNbr` is an arbitrary string entered by users (it should not be limited to a number/integer, because of roman numerals such as `ii` - but I am filing a separate issue for this). So `pageNbr` must be cleaned up as well, so that it can be compared with `page.Title`.

Reply:
I changed the pageNbr. It's now a string

Reply:
Yes, but it is not normalized (leading+trailing trim of whitespace characters, and inner whitespace collapsing).
Once it is normalized, than it can be compared reliably with the base `pageLink.Title`

Reply:
I added the same normalization